src/libcore/mem.rs

```diff
@@ -611,3 +611,38 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
```

Comment:
I think we should say that consumers shouldn't depend on discriminants being stable across versions of a dependency unless otherwise documented on the enum.

I think it's probably reasonable to state that discriminants will be stable for a given enum definition though across compilations. Maybe compiler versions as well?